src/style-property-map.js

```diff
@@ -20,7 +20,26 @@
   internal.inherit(StylePropertyMap, internal.StylePropertyMapReadOnly);
 
   StylePropertyMap.prototype.set = function(property, value) {
-    throw new TypeError('Function not implemented yet');
+    var cssPropertyDictionary = propertyDictionary();
+    if (!cssPropertyDictionary.isSupportedProperty(property)) {
+      throw new TypeError(property + 'Is not a Supported CSS property');
+    }
+
+    if (value instanceof StyleValue) {
+      if (!cssPropertyDictionary.isValidInput(property, value)) {
+        throw new TypeError(
+          'This StyleValue type cannot be set to this property');
+      }
+      this._styleObject[property] = value.cssString;
+      return;
+    }
+
+    if (value instanceof Array) {
```

Comment:
How about we put this first, then it can eventually be passed off to a helper function. Then you can get rid of the floating `return` statement and write something like

```javascript
StylePropertyMap.prototype.set = function(property, value) {
  var cssPropertyDictionary = propertyDictionary();
  if (!cssPropertyDictionary.isSupportedProperty(property)) {
    throw new TypeError(property + ' is not a supported CSS property.');
  }

  if (value instanceof Array) {
    // pass to helper function
  }

  if (!value instanceof StyleValue) {
    throw new TypeError('value must be a StyleValue or sequence of StyleValues');
  }
  if (!cssPropertyDictionary.isValidInput(property, value)) {
    throw new TypeError(property + ' does not take values of type ' + value.constructor.name);
  }

  this._styleObject[property] = value.cssString;
}
```